picamera2/picamera2.py

```diff
@@ -455,9 +532,9 @@ def stop_preview(self) -> None:
         """
         if self._preview:
             try:
+                self.have_event_loop = False
```

Comment:
Was there a reason for moving this line up? Doesn't matter, I was just curious!

Reply:
I might have been testing something after setting it, but can't recall afraid.